config/main.py

```diff
@@ -331,6 +331,16 @@ def _stop_services():
             log_error("Stopping {} failed with error {}".format(service, e))
             raise
 
+    # iccpd service doesn't start by default
+    (out, err) = run_command("systemctl status iccpd", return_output = True)
```

Comment:
What is the mechanism for enabling the iccpd service? We recently added a new global mechanism for enabling/disabling optional features (where each feature is a Docker container managed by a systemd service). It appears that iccpd is a great candidate for this. I believe iccpd should be handled using the global mechanism and the changes in this file should be made to walk all optional services and stop/reset-failed/restart them as appropriate.

Reply:
Currently we use `systemctl [enable|disable] iccpd` to enable/disable iccpd service. It seems that we can leverage `config iccpd {enable|disable}` to do it. Per your suggestion we can add `optional_services_to_reset` list to walk all optional services. How do you think it? @jleveque